opencti-platform/opencti-front/src/components/BulkAddComponent.tsx

```diff
@@ -0,0 +1,102 @@
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogActions from '@mui/material/DialogActions';
+import Typography from '@mui/material/Typography';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import React, { useEffect } from 'react';
+import { Field } from 'formik';
+import { useFormatter } from './i18n';
+import TextField from './TextField';
+
+type BulkAddComponentProps = {
+  openBulkModal: boolean
+  bulkValueFieldValue: string
+  handleOpenBulkModal: () => void
+  handleCloseBulkModal: (val: string) => void
+  localHandleCancelClearBulkModal: () => void
+};
+
+const BulkAddComponent: React.FC<BulkAddComponentProps> = ({
+  openBulkModal,
+  bulkValueFieldValue,
+  handleOpenBulkModal,
+  handleCloseBulkModal,
+  localHandleCancelClearBulkModal,
+}) => {
+  const { t_i18n } = useFormatter();
+  const [localBulkValueField, setLocalBulkValueField] = React.useState('');
+  const [warningVisible, setWarningVisible] = React.useState(false);
+
+  function monitorBulkValue(value: string) {
+    const observable_lines = value.split('\n');
+    // 50 lines have been entered based on \n newlines - warn user and remove Continue button.
+    if (observable_lines.length >= 51) {
+      if (warningVisible === false) {
+        setWarningVisible(true);
+      }
+    } else if (warningVisible === true) {
+      setWarningVisible(false);
+    }
```

Comment:
```suggestion
    if (observable_lines.length >= 51 !== warningVisible) {
      setWarningVisible(observable_lines.length >= 51);
    }
```

Reply:
Completed in latest commit. Accepted suggested change. 47b5d44b47d2bf9e61c473dd6e7d1bd9a8a1c365